spec/api_callers/http_request_spec.rb

```diff
@@ -2,22 +2,36 @@
 require 'net/http'
 
 describe ::ApiCallers::HttpRequest do
-  subject { ::ApiCallers::HttpRequest.new('') }
+  subject { ::ApiCallers::HttpRequest.new('some_uri', http_verb) }
+  let(:http_verb) { 'get' }
 
-  describe "#headers" do
-    it "returns { 'Accept-Charset' => 'utf-8' }" do
-      expect(subject.send(:headers)).to eq({ 'Accept-Charset' => 'UTF-8' })
+  describe "#make_request" do
+    let(:http_obj) { double('http', request: request) }
+    let(:request) { double('request', body: { response: 'response'}) }
+    let(:url) { double('url', request_uri: 'request_uri', host: 'host', port: 'port') }
+
+    before do
+      allow(URI).to receive(:parse).and_return(url)
+      allow(Net::HTTP).to receive(:new).and_return(http_obj)
     end
-  end
 
-  describe "#class_to_call" do
-    it "returns Net::HTTP::Get by default" do
-      expect(subject.send(:class_to_call)).to eq Net::HTTP::Get
+    context 'get request' do
+      it 'make a request to the provided url' do
+        allow(Net::HTTP::Get).to receive(:new).with('request_uri', { 'Accept-Charset' => 'UTF-8' }).and_return(:get_request)
```

Comment:
Line is too long. [124/80]<br>Redundant curly braces around a hash parameter.<br>Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.